### PR TITLE
feat: 100% mantine

### PR DESCRIPTION
**Pull request by IrakliJani**
### Description:

Final PR to remove Blueprint from the project!

### Summary

- Removed the `@blueprintjs/core` npm package.
- Deleted the `blueprint-core.css` file.
- Eliminated `BlueprintProvider` and moved some CSS globals to `mantineTheme`.
- Cleaned up `mantineTheme` by removing Blueprint-related items.
- Removed the Blueprint counter GitHub workflow from `.github/workflows/count-blueprint`.
- Deleted the Blueprint import counter script `scripts/count-blueprint-imports.ts`.

After these changes, the build was failing. The exact cause was unclear, but @magnew assisted in identifying the issue related to Vite build and [module chunking](https://vitejs.dev/guide/build.html#chunking-strategy). Removing the Blueprint chunk likely disrupted the module resolver's hierarchy and order. Suspecting a bug with Rollup, I decided to upgrade to the latest Vite version, which includes the most recent Rollup version.

I made following changes to Vite:

### Changes to VITE

- Upgraded Vite to the latest version.
- Updated Vite plugins to their latest versions.
- Resolved an issue with SVGR TS types; the Vite SVGR plugin now requires a `?react` suffix in imports to handle SVGs as React components.
- Removed ESLint from Vite, as it's not necessary and was slowing down the dev mode.
- Switched from lodash-es to lodash after realizing both were included in the build.
- Implemented new and improved module chunks.

### New Module Chunks
```
react:      ['react', 'react-dom', 'react-router-dom', 'react-hook-form', 'react-use', 'react-query', 'react-beautiful-dnd', 'react-draggable', '@tanstack/react-virtual', '@tanstack/react-table'],
echarts:    ['echarts'],
vega:       ['vega', 'vega-lite'],
ace:        ['ace-builds', 'react-ace/lib'],
modules:    ['moment/moment.js', 'moment/dist/moment.js', 'pegjs', 'jspdf', 'ajv', 'ajv-formats', 'lodash', 'colorjs.io', 'liquidjs', 'zod'],
thirdparty: ['@sentry/react', '@sentry/tracing', 'rudder-sdk-js', 'posthog-js'],
uiw:        ['@uiw/copy-to-clipboard', '@uiw/react-markdown-preview', '@uiw/react-md-editor'],
mantine:    ['@mantine/core', '@mantine/dates', '@mantine/form', '@mantine/hooks', '@mantine/notifications', '@mantine/prism', '@mantine/spotlight'],
```

CleanShot 2023-12-21 at 22 49 07@2x

The goal was to keep chunks below 500kb (gzipped) and logically grouped.

Here's how `npx vite-bundle-visualizer` displays the chunks (FYI, they are color-coded):
CleanShot 2023-12-21 at 22 50 47@2x

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/ReactHookForm/InputWrapper.css
Comment (magnew) [inline packages/frontend/src/components/ReactHookForm/InputWrapper.css:1]: 🎉

**Review by magnew — APPROVED** [review]
Woohoo! Approved for whats here. Happy to look again at the build fix, but I don't wanna block you. 

There is one move Blueprint FIXME in PageHeader. We can add it here or make another review.

File: packages/frontend/src/mantineTheme.ts [hunk 1]
```diff
@@ -145,6 +74,7 @@ export const getMantineThemeOverride = (overrides?: {
                 yOffset: 140,
```
Comment (magnew) [inline packages/frontend/src/mantineTheme.ts:74]: There's this one too

File: packages/frontend/src/MobileRoutes.tsx [hunk 2]
```diff
@@ -41,7 +41,7 @@ import MobileSpaces from './pages/MobileSpaces';
 import Projects from './pages/Projects';
 import ShareRedirect from './pages/ShareRedirect';
 import { TrackPage } from './providers/TrackingProvider';
-import { ReactComponent as Logo } from './svgs/logo-icon.svg';
+import Logo from './svgs/logo-icon.svg?react';
```
Comment (magnew) [inline packages/frontend/src/MobileRoutes.tsx:44]: Different syntax for vite or for something else?

File: packages/frontend/vite.config.ts [hunk 3]
```diff
@@ -47,17 +29,55 @@ export default defineConfig({
...
-                    '@blueprintjs/': 'blueprint-vendor',
-                    '@mantine': 'mantine-vendor',
-                    'highlight.js': 'highlight-vendor',
-                    echarts: 'echarts-vendor',
-                    '@mapbox/': 'mapbox-vendor',
-                    rudder: 'rudder-vendor',
-                    sentry: 'sentry-vendor',
-                    'react-vega': 'react-vega-vendor',
-                }),
+                manualChunks: {
```
Comment (magnew) [inline packages/frontend/vite.config.ts:32]: Boom! Extra credit. Nice work here -- the sizes are much better now.

**Review by magnew — APPROVED** [review]
Looks great to me. The new chunking is great and it's good you got some of these things updated. Nice work! 💥

I checked out and built clean locally and tested. I don't see any issues.

File: packages/frontend/src/MobileRoutes.tsx (hunk 2 above)
Comment (IrakliJani) [reply packages/frontend/src/MobileRoutes.tsx:44]: yeah, it's an svgr plugin upgrade thing

Comment (IrakliJani) [reply packages/frontend/src/MobileRoutes.tsx:44]: if you omit `?react` it will import as asset (uri string)

**Pull request merged**